sw/host/tests/manuf/provisioning/cp_test/src/main.rs

```diff
@@ -27,24 +179,35 @@ fn main() -> Result<()> {
     opts.init.init_logging();
     let transport = opts.init.init_target()?;
 
+    // Generate random provisioning values for testing.
+    let mut device_id = ArrayVec::<u32, 8>::new();
+    let mut manuf_state = ArrayVec::<u32, 8>::new();
+    let mut wafer_auth_secret = ArrayVec::<u32, 8>::new();
+    let mut test_exit_token = ArrayVec::<u32, 4>::new();
+    let mut test_unlock_token = ArrayVec::<u32, 4>::new();
```

Comment:
nit: Since these `ArrayVec`s are moved into the `ManufCpProvisioningData`, the compiler should be able to deduce the generic type parameters here (ie: `<u32, 8>`, etc).

If you feel that the code is clearer with the parameters explicitly specified, then I'd live things as-is.

Reply:
ah neat, fixed.